cypress/e2e/api/engageShere.cy.js

```diff
@@ -0,0 +1,144 @@
+describe('Listar clientes na aplicação EngageSphere', () => {
+  const CUSTOMERS_API_URL = `${Cypress.env('API_URL')}`
+  it('Recuperar clientes com sucesso', () => {
+    cy.request('GET', `${CUSTOMERS_API_URL}customers?page=1&limit=10&size=All`).then(
+      (response) => {
+        expect(response.status).to.eq(200)
+      }
+    )
+  })
+
+  it('Validar filtro pelo tamanho da empresa', () => {
+    const tamanhos = [
+      'Small',
+      'Medium',
+      'Enterprise',
+      'Large Enterprise',
+      'Very Large Enterprise',
+    ]
+
+    tamanhos.forEach((tamanho) => {
+      cy.request('GET',`${CUSTOMERS_API_URL}customers?page=1&limit=10&size=${tamanho}`
+      ).then(({ body }) => {
+        body.customers.forEach((customer) => {
+          expect(customer.size).to.eq(tamanho)
+        })
+      })
+    })
+  })
+
+  it('Validar filtro por paginação', () => {
+    const qtdaPaginas = ['5', '10', '20', '50']
+
+    qtdaPaginas.forEach((paginas) => {
+      cy.request('GET', `${CUSTOMERS_API_URL}customers?page=1&limit=${paginas}&size=All`
+      ).then(({ body }) => {
+        body.pageInfo.forEach((page) => {
+          expect(page.totalPages).to.have.length(paginas)
+        })
+      })
+    })
+  })
+
+  it('Validar as propriedades customer e pagInfo da resposta', () => {
+    cy.request('GET', `${CUSTOMERS_API_URL}customers?page=1&limit=5&size=All`).then(
+      ({ body }) => {
+        expect(body).to.have.property('customers')
+        expect(body).to.have.property('pageInfo')
+        expect(body.customers).to.be.an('array')
+        expect(body.customers.length).to.be.greaterThan(0)
+
+        body.customers.forEach((customer) => {
+          expect(customer).to.have.property('id')
+          expect(customer).to.have.property('name')
+          expect(customer).to.have.property('employees')
+          expect(customer).to.have.property('size')
+          expect(customer).to.have.property('address')
```

Comment:
Faltou a seguinte verificação: `expect(customer).to.have.property('contactInfo')`